### PR TITLE
Implements a new interface for `create`

### PR DESCRIPTION
**Pull request by miguelgondu**
## Main change

Previously, `create` returned 5 elements (of which usually a single one was relevant: `f`). For example:

```python
from poli import create

problem_info, f, x0, y0, run_info = create(name="white_noise)
```

Of these, `problem_info` was already inside the black box as `f.info`, and `run_info` was only relevant when an observer was passed.

This PR modifies the interface such that `create` returns only `f, x0, y0`. The `run_info` is now stored as `f.observer_info`. Backwards compatibility can be achieved by running:

```python
from poli import create

# Creation of f with observer
f, x0, y0 = create(name="white_noise, observer=...)

problem_info, run_info = f.info, f.observer_info
```

## Other changes

I changed the names of some kwargs to make them more transparent: now `caller_info` is called `observer_init_info`

### CODE REVIEW

File: src/poli/objective_factory.py [hunk 1]
```diff
@@ -420,18 +416,17 @@ def create(
...
         )
-        problem_info = f.info
 
-        observer_info = None
         if observer is not None:
             if not quiet:
                 print(f"poli 🧪: initializing the observer.")
             observer_info = observer.initialize_observer(
-                problem_info, caller_info, x0, y0, seed
+                f.info, observer_init_info, x0, y0, seed
```
Comment (RMichae1) [inline src/poli/objective_factory.py:424]: LGTM

**Review by RMichae1 — APPROVED** [review]
LGTM!

File: src/poli/core/abstract_black_box.py
Comment (SimonBartels) [inline src/poli/core/abstract_black_box.py:1]: Is there a reason why the objective function needs to know about its observer? I'd expect that this just opens the door for shenanigans.

Comment (miguelgondu) [reply src/poli/core/abstract_black_box.py:1]: I don't think it opens to any more shenanigans than having access to the observer itself, right? Maybe I'm missing something.

Comment (RMichae1) [reply src/poli/core/abstract_black_box.py:1]: Is this part of this current feature refactor?  
Or is this part of another API change (dedicated issue) where we address observers specifically?

To me the proposed API change is about the `poli` `create` return tuple.
I don't think we're trying to solve the encapsulation of the observers in this particular PR.

For my understanding: we can still use an external observer in the current setup and separate as has previously been the case.

Comment (miguelgondu) [reply src/poli/core/abstract_black_box.py:1]: I agree with Richard: that's a separate issue that we can address on its own.

@SimonBartels, do you have any comments on the API change in particular?

Comment (SimonBartels) [reply src/poli/core/abstract_black_box.py:1]: Ah, I misread the change in `AbstractBlackbox`. The object just gets the info returned by the observer not the observer itself. Sorry, my bad. I am okay with that.

**Pull request merged**